homeassistant/components/vrm_forecasts/sensor.py

```diff
@@ -0,0 +1,248 @@
+    ),
+    VRMForecastsSensorEntityDescription(
+        key="energy_consumption_current_hour",
+        translation_key="energy_consumption_current_hour",
+        state=lambda estimate: estimate.consumption.current_hour_total,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=1,
+    ),
+    VRMForecastsSensorEntityDescription(
+        key="energy_consumption_next_hour",
+        translation_key="energy_consumption_next_hour",
+        state=lambda estimate: estimate.consumption.next_hour_total,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=1,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: VRMForecastsConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = entry.runtime_data
+
+    async_add_entities(
+        VRMForecastsSensorEntity(
+            entry_id=entry.entry_id,
+            coordinator=coordinator,
+            entity_description=entity_description,
+        )
+        for entity_description in SENSORS
+    )
+
+
+class VRMForecastsSensorEntity(
+    CoordinatorEntity[VRMForecastsDataUpdateCoordinator], SensorEntity
+):
+    """Defines a VRM Solar Forecast sensor."""
+
+    entity_description: VRMForecastsSensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        *,
+        entry_id: str,
+        coordinator: VRMForecastsDataUpdateCoordinator,
+        entity_description: VRMForecastsSensorEntityDescription,
+    ) -> None:
+        """Initialize VRM Solar Forecast sensor."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = entity_description
+        self.entity_id = f"{SENSOR_DOMAIN}.{entity_description.key}"
+        self._attr_unique_id = (
+            f"{entry_id}_{entity_description.key}_{coordinator.data.site_id}"
+        )
+
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer="Victron Energy",
+            model=f"VRM - {coordinator.data.site_id}",
+            name="VRM Forecast",
+            configuration_url="https://vrm.victronenergy.com",
+        )
+
+    @property
+    def native_value(self) -> datetime | StateType:
+        """Return the state of the sensor."""
+        if self.entity_description.state is None:
+            state: StateType | datetime = getattr(
+                self.coordinator.data, self.entity_description.key
+            )
+        else:
+            state = self.entity_description.state(self.coordinator.data)
```

Comment:
Let's not use getattr, instead, use `state` everywhere and just access it normally. This way it's also type safe